src/tribler-core/tribler_core/modules/process_checker.py

```diff
@@ -56,7 +56,7 @@ def create_lock_file(self):
         might be called before the .Tribler directory has been created.
         """
         if not self.state_directory.exists():
-            path_util.makedirs(self.state_directory)
+            pathlib.Path(self.state_directory).mkdir(parents=True)
```

Comment:
Try making `self.state_directory` into a Path object in `__init__` and where it is set/used.

Reply:
Done